### ISSUE TITLE
Reconsider default huge page requirement for compute and memory IT classes

### ISSUE THREAD

**Issue by cwilkers** [kubevirt/common-instancetypes#105]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
/kind enhancement

**What happened**:

Using a compute or memory instance type requires a node with huge pages support, which is not a default setting, so out of the box, compute and memory instance types are unusable.

When trying to create a VM in one of these instance types, it fails with an unhelpful error that points to scheduling, but does not mention huge pages as the reason.

**What you expected to happen**:

Either better messaging to bring to light the lack of huge pages or another way to introduce to admins the ability to require huge pages in an instance type.

**How to reproduce it (as minimally and precisely as possible)**:

Create a vm with compute or memory instance type on a cluster where nodes do not have huge page support.

**Anything else we need to know?**:

Huge pages make sense across the board for VM handling nodes, but fine tuning the number of huge pages per compute node is a per-cluster exercise, and fits better in a performance and tuning guide than a default.

**Comment by lyarwood** [comment]
I agree that `hugepages` likely doesn't make sense on the compute intensive class but disagree that the error shown on the `VirtualMachine` and `VirtualMachineInstance` are somehow unclear to users.

We expose requirements through labels currently but should also better document these in the description annotation with links to upstream documentation. Vendors could easily rewrite these downstream with downstream links so I really don't mind adding these.

@fabiand @cwilkers  @0xFelix  Thoughts?

**Comment by cwilkers** [comment]
I know that when I'm looking at that message, my first interpretation is that preemption is the culprit, and I'm wondering what that even means. I have heard from users on kubevirt-dev that they see the message about cloudinitdisk's lack of snapshot as the error being presented to them, so without a more direct message, any bit of text there can lead somebody on a wild goose chase.

**Comment by lyarwood** [comment]
> I know that when I'm looking at that message, my first interpretation is that preemption is the culprit

It's likely easier to gork broken down into the initial scheduling failure reasons and then the follow up part about preemption:

- 0/6 nodes are available: 
  - 1 node(s) were unschedulable
  - 2 Insufficient hugepages-2Mi
  - 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. 

 - preemption: 0/6 nodes are available: 
   - 2 No preemption victims found for incoming pod
   - 4 Preemption is not helpful for scheduling..

> and I'm wondering what that even means. 

https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#pods-are-preempted-unnecessarily

tl;dr [Pods](https://kubernetes.io/docs/concepts/workloads/pods/) can have priority. Priority indicates the importance of a Pod relative to other Pods. If a Pod cannot be scheduled, the scheduler tries to preempt (evict) lower priority Pods to make scheduling of the pending Pod possible.

> I have heard from users on kubevirt-dev that they see the message about cloudinitdisk's lack of snapshot as the error being presented to them, so without a more direct message, any bit of text there can lead somebody on a wild goose chase.

I think the `ErrorUnschedulable` `Printable Status` makes this abundantly clear to users what the issue is. Yes there's lots of details also listed that might trip up less seasoned k8s users and operators but that isn't something I think KubeVirt can really impact tbh.

IMHO it's a million times better than the scheduling errors we surfaced in `openstack/nova`, `no valid hosts` was so damn awful until we started providing more context to admins.

For this part if you still feel strongly enough please raise an issue against core [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt/issues) as this isn't something we can address in this repo.

**Comment by 0xFelix** [comment]
`Message:               0/6 nodes are available: 1 node(s) were unschedulable, 2 Insufficient hugepages-2Mi, 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. preemption: 0/6 nodes are available: 2 No preemption victims found for incoming pod, 4 Preemption is not helpful for scheduling..`

I've always found this message very complicated to understand because it packs a lot of information into one line of text, but basically the information is there. As @lyarwood said, maybe it can be addressed in [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt/issues), but unfortunately not here.

Regarding the issue of unusable compute and memory instance types out of the box, we should make it obvious that for default use cases the universal instance types should be used. Compute and memory intensive instance types are more performance orientated and need clusters with respective resources.
